src/bin/sync-categories.rs

```diff
@@ -0,0 +1,50 @@
+// Sync available crate categories from `src/categories.txt`.
+// Only needs to be run for new databases or when `src/categories.txt`
+// has been updated.
+//
+// Usage:
+//      cargo run --bin sync-categories
+
+#![deny(warnings)]
+
+extern crate cargo_registry;
+extern crate postgres;
+
+use cargo_registry::env;
+
+fn main() {
+    let conn = postgres::Connection::connect(&env("DATABASE_URL")[..],
+                                             postgres::TlsMode::None).unwrap();
+    {
+        let tx = conn.transaction().unwrap();
+        sync(&tx).unwrap();
+        tx.set_commit();
+        tx.finish().unwrap();
+    }
+}
+
+fn sync(tx: &postgres::transaction::Transaction) -> postgres::Result<()> {
+    let categories = include_str!("../categories.txt");
+    let categories = categories.lines().collect::<Vec<_>>();
```

Comment:
I prefer `let categories: Vec<_> = categories.lines().collect();`, unless this is a crates.io convention.